sotrplib/sources/sources.py

```diff
@@ -1,4 +1,252 @@
-from pydantic import BaseModel
+from typing import Any, Literal, Optional
+
+import structlog
+from astropy import units as u
+from astropydantic import AstroPydanticQuantity
+from pydantic import BaseModel, Field
+from structlog.types import FilteringBoundLogger
+
+
+class BaseSource(BaseModel):
+    ra: AstroPydanticQuantity[u.deg]
+    dec: AstroPydanticQuantity[u.deg]
+    flux: Optional[AstroPydanticQuantity[u.mJy]] = None
+
+
+class RegisteredSource(BaseSource):
+    """
+    A registered source object which may have catalog crossmatches.
+    This object has a source_id which may belong to the catalog and a source type.
+
+    This object need not have a flux.
+
+    Extendedness and positional uncertainty is stored if available.
+    """
+
+    source_id: str | None = None
+    source_type: Literal["Extragalactic", "Star", "Asteroid", "Unknown"] | None = None
+
+    crossmatch_names: list | None = None
+    crossmatch_probabilities: list | None = None
+    crossmatch_fluxes: list | None = None
+    crossmatch_frequencies: list | None = None
```

Comment:
Instead of None, why not set them to []?

Reply:
Alternatively, instead of dealing with trying to keep the lists consistent, why not just have
```
class CrossMatch(BaseModel):
  name: str
  probability: float
  flux: AstroPydanticQuantity[u.Jy]
  frequency: AstroPydanticQuantity[u.Hz]
```
And then have
```
crossmatches: list[CrossMatch]
```